lib/lib.go

```diff
@@ -661,19 +679,29 @@ func (inst *Instance) Connect(ctx context.Context) (err error) {
 	// old instance, we run into issues where the online instance can't "see"
 	// the additions. We fix that by re-initializing the client and remote with the new
 	// instance
-	if inst.remoteClient, err = remote.NewClient(inst.node); err != nil {
+	if inst.remoteClient, err = remote.NewClient(ctx, inst.node); err != nil {
```

Comment:
this old ipfs node instance reference business is so hairy.

Trying to think of ways we can refactor in the future so it doesn't smell so much. Should `remoteClient`, `remote`, & `registry` really be the responsibility of the `node` (which you can argue exists to manage the communications btw this instance and other instances.) Then the `remoteClient` & `remote` update can exist closer to where the actual ipfs node replacement happens.

Should `remoteClient` or `remote` only initialize right before they are used?